src/datatype/aggregate_svecf32.rs

```diff
@@ -0,0 +1,485 @@
+    pub fn merge_in_place(&mut self, svec: SVecf32Borrowed<'_>) -> u32 {
+        let sindexes = svec.indexes();
+        let svalues = svec.values();
+        let slen = sindexes.len();
+        let len = self.len;
+        let capacity = self.capacity as usize;
+        // To reduce the memory usage, we try to merge the sparse vector in place.
+        // If the state capacity is enough, we can merge the sparse vector from the end. Then copy the result to the beginning. The merge process compares the indexes of the sparse vector and the state from end to start, and the result is stored in the state from the end to start to avoid the conflict.
+        let mut i = (len as i32) - 1;
+        let mut j = (slen as i32) - 1;
+        let mut p = (capacity as i32) - 1;
+        while i >= 0 && j >= 0 {
+            let index = self.indexes[i as usize];
+            let sindex = sindexes[j as usize];
+            let value = self.values[i as usize];
+            let svalue = svalues[j as usize];
+            let pi = std::cmp::max(index, sindex);
+            let pv = F32((pi == index) as usize as f32) * value
+                + F32((pi == sindex) as usize as f32) * svalue;
+            i -= (index >= sindex) as i32;
+            j -= (index <= sindex) as i32;
+            assert!(p > i, "Conflict occurs when merge in place.");
+            self.indexes[p as usize] = pi;
+            self.values[p as usize] = pv;
+            // Skip zero value.
+            p -= (!pv.is_zero()) as i32;
+        }
+        while j >= 0 {
+            assert!(p > i, "Conflict occurs when merge in place.");
+            self.indexes[p as usize] = sindexes[j as usize];
+            self.values[p as usize] = svalues[j as usize];
+            p -= 1;
+            j -= 1;
+        }
+        self.len = if i < 0 {
+            // move the whole state [p+1..capacity] to the beginning
+            self.indexes.copy_within(((p + 1) as usize)..capacity, 0);
+            self.values.copy_within(((p + 1) as usize)..capacity, 0);
+            ((capacity as i32) - p - 1) as u32
+        } else {
+            // concatenate the state [p+1..capacity] and remaining state [0..i]
+            self.indexes
+                .copy_within(((p + 1) as usize)..capacity, (i as usize) + 1);
+            self.values
+                .copy_within(((p + 1) as usize)..capacity, (i as usize) + 1);
+            ((capacity as i32) - p + i) as u32
+        };
+        self.len
+    }
+}
+
+/// accumulate intermediate state for sparse vector
+#[base_macros::aggregate_func]
+#[pgrx::pg_extern(immutable, parallel_safe)]
+fn _vectors_svecf32_aggregate_avg_sum_sfunc(
+    mut current: Option<Internal>,
+    value: Option<SVecf32Input<'_>>,
+) -> Option<Internal> {
+    if value.is_none() {
+        // It would get error "returned Datum was NULL" if we use Internal directly when execute `SELECT sum(v) FROM unnest(ARRAY[NULL]::svector[]) v;`
```

Comment:
It seems upstream have solve it. https://github.com/pgcentralfoundation/pgrx/pull/1701

Reply:
You could submit a PR to use branch `v0.12.0-alpha.1-patch2`.